src/bin/pg_autoctl/pgsetup.c

```diff
@@ -367,33 +375,13 @@ pg_setup_init(PostgresSetup *pgSetup,
 		 */
 		if (!pgIsReady)
 		{
-			log_error("Failed to read Postgres pidfile, see above for details");
-			return false;
-		}
-
-		/*
-		 * Postgres is running, is it in recovery?
-		 *
-		 * We're going to connect to "template1", because our target database
-		 * might not have been created yet at this point: for instance, in case
-		 * of problems during the `pg_autoctl create` command.
-		 */
-		strlcpy(dbname, pgSetup->dbname, NAMEDATALEN);
-		strlcpy(pgSetup->dbname, "template1", NAMEDATALEN);
-
-		/* initialise a SQL connection to the local postgres server */
-		pg_setup_get_local_connection_string(pgSetup, connInfo);
-		pgsql_init(&pgsql, connInfo, PGSQL_CONN_LOCAL);
-
-		if (!pgsql_is_in_recovery(&pgsql, &pgSetup->is_in_recovery))
```

Comment:
Why do we not need this anymore?

Reply:
Exactly. It's been there since about the beginning, but we moved a long time ago already to using other means to update that parts, see `pgsql_get_postgres_metadata` that combines several pieces of information that we need in a single query: https://github.com/citusdata/pg_auto_failover/blob/master/src/bin/pg_autoctl/pgsql.c#L1664